### PR TITLE
Add HG-3 and HG-3-SL

### PR DESCRIPTION
**Pull request by Ash19256**
Adding two historical engines that IRL were stepping stones between the J-2 and the RS-25 aka Space Shuttle Main Engine.

### CODE REVIEW

File: GameData/RealismOverhaul/RO_SuggestedMods/SXT/RO_SXT_Engines.cfg [hunk 1]
```diff
@@ -167,6 +167,107 @@
 	}
 	engineType = J2
 }
+// HG-3
++PART[SXTLT80]:FOR[RealismOverhaul]
```
Comment (raidernick) [inline GameData/RealismOverhaul/RO_SuggestedMods/SXT/RO_SXT_Engines.cfg]: should be changed to +PART[SXTLT80]:BEFORE[RealismOverhaul]

File: GameData/RealismOverhaul/RO_SuggestedMods/SXT/RO_SXT_Engines.cfg [hunk 2]
```diff
@@ -167,6 +167,107 @@
...
+		name = ModuleGimbal
+		gimbalTransformName = gimbal
+		gimbalRange = 7.5
+		useGimbalResponseSpeed = true
+		gimbalResponseSpeed = 16
+	}
+	engineType = HG3
+}
+// HG-3-SL
++PART[SXTLT80]:FOR[RealismOverhaul]
```
Comment (raidernick) [inline GameData/RealismOverhaul/RO_SuggestedMods/SXT/RO_SXT_Engines.cfg]: change to +PART[SXTLT80]:BEFORE[RealismOverhaul]

**Comment by Ash19256** [comment]
Changes made

**Comment by PhineasFreak** [comment]
BTW, the TweakScale stuff for the engine parts is not required anymore, as we have a global patch to take care of that: https://github.com/KSP-RO/RealismOverhaul/blob/master/GameData/RealismOverhaul/RO_Engines.cfg#L17-L25

**Comment by Ash19256** [comment]
Acknowledged, removed it to ensure there was no interference with the global patch.

File: GameData/RealismOverhaul/RO_SuggestedMods/SXT/RO_SXT_Engines.cfg [hunk 3]
```diff
@@ -167,6 +167,105 @@
 	}
 	engineType = J2
 }
+// HG-3
++PART[SXTLT80]:BEFORE[RealismOverhaul]
+{
+	@name=HG-3
+	%RSSROConfig = True
+	{
+	}
```
Comment (Olympic1) [inline GameData/RealismOverhaul/RO_SuggestedMods/SXT/RO_SXT_Engines.cfg]: Module was deleted in https://github.com/KSP-RO/RealismOverhaul/pull/1609/commits/90d1cb68280406f80f2e82ddbe25ad6e73fac924 but you forgot the brackets

File: GameData/RealismOverhaul/RO_SuggestedMods/SXT/RO_SXT_Engines.cfg [hunk 4]
```diff
@@ -167,6 +167,105 @@
...
+	engineType = HG3
+}
+// HG-3-SL
++PART[SXTLT80]:BEFORE[RealismOverhaul]
+{
+	@name=HG-3-SL
+	%RSSROConfig = True
+	{
+	}
```
Comment (Olympic1) [inline GameData/RealismOverhaul/RO_SuggestedMods/SXT/RO_SXT_Engines.cfg]: same here

**Comment by Ash19256** [comment]
Thank you for pointing that out.

**Comment by Ash19256** [comment]
Well, now that I know that engine configs are a thing, would it be a good idea for me to add the HG-3 and HG-3-SL to the J-2 series engine configs? The HG-3 and HG-3-SL were derived from the J-2, and have higher performance.

**Comment by NathanKell** [comment]
No. They're different engines, not upgrades.

That'd be like putting HG-3 and RS-25 configs on the same part.

**Comment by Ash19256** [comment]
Changed things so that I made use of the Engine Config system, thanks to advice from Tyler Raiz.

**Comment by Ash19256** [comment]
Created HG-3 engine and HG-3 Config.

File: GameData/RealismOverhaul/Engine_Configs/HG-3_Config.cfg [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+	@title = HG-3 Series
+	%manufacturer = Rocketdyne
+	@description = Rocketdyne HG-3 engine, which served as a stepping stone between the J-2 and RS-25. Ensure you enable roll capability or disable gimbal as required for your application. Diameter: [2.01 m].
+	
+	MODULE
+	{
+		name = ModuleEngineConfigs
+		type = ModuleEngines
+		configuration = J-2
+		@configuration:NEEDS[RP-0] = J-2-200klbf
```
Comment (SirKeplan) [inline GameData/RealismOverhaul/Engine_Configs/HG-3_Config.cfg]: configuration needs changing to HG-3

and the `@configuration:NEEDS` line can probably go

**Comment by SirKeplan** [comment]
Would it make sense to give the upper stage variant extra ignitions? considering it was considered for use on the Saturn MLV's modified S-IVB stage.

**Comment by Ash19256** [comment]
Right, so this is basically done, unless anyone like @raidernick or @NathanKell has anything else they need me to add/change.

**Comment by PhineasFreak** [comment]
@Ash19256 Copying the J-2 RP config is a fine way to start making one for these engines. Make sure that the names of the RP patches match the part config name ("HG-3").

Also, minor request: can you change the name of the part from "HG-3" to "RO-HG-3" or something similar? Just to make sure that we won't have any naming conflicts with other parts down the road.

**Comment by Ash19256** [comment]
Name changed. Looking at the Real Plume configs, unless there is some sort of universal engine plume config ala the EngineConfigs system, the SXT LT80 which is being used as the base of these parts already has a plume configured for it by RO, so I'm not sure if this still actually needs a plume configured for it. If it does, is there a tutorial for that someone can link?

**Comment by pap1723** [comment]
https://www.youtube.com/watch?v=sNJubaaMTTE
Made by @stratochief66

**Comment by Ash19256** [comment]
... I have the distinct problem that I can't actually run RO with all of the mods I would need to test this. So I'm probably going to need to get someone else to help with that part. I figure the Hydrolox-Upper config that is attached to the SXT LT80 part already will work well enough until such time as someone can make a better configuration.

**Comment by Ash19256** [comment]
So, assuming that the predicted plume issue isn't an issue, is this good as is?

**Comment by SirKeplan** [comment]
Yeah you need to add a plume config, it has no running effects at all ingame right now. a lightly modified copy of this file should do it. https://github.com/KSP-RO/RealismOverhaul/blob/master/GameData/RealismOverhaul/RealPlume_Configs/SXT/SXTLT80.cfg

Another thing, i'm quite skeptical of astronautix's claim that the sl variant reaches 450s Isp in vacuum, seems very unusual for a presumably low expansion engine. havent found any other source for this though.

**Comment by stratochief66** [comment]
@SirKeplan you were active in this. Other than the plume, does this appear to be ok to merge? The plume could be added later, if it doesn't seem to be forthcoming.

**Comment by Ash19256** [comment]
Well, I've been in contact with the YouTuber Cosmonaut Crash, and about the only things that need work is that one the TestFlight part of this whole thing doesn't work for some reason, and the plume still needs to be finished. He's been working on the plume, so I'll see about getting him to toss the file for that my way so I can merge it with this. As for the TestFlight thing, I'm as clueless as he is on the subject. He's been saying that TestFlight flat out isn't able to detect these engines at all, which is weird, and I have no idea why. I copied the TestFlight bits from the J-2 and NK-15 with regards to formatting and stats, so I'm not sure why it isn't working at all.

**Comment by rsparkyc** [comment]
Does the ModuleManager.ConfigCache show the testflight bits being applied?

**Comment by Ash19256** [comment]
From what I recall from the file I was sent when we were trying to
troubleshoot the engine, yeah, both configs received TestFlight bits. They
just weren't doing anything. I'll link you guys to the dropbox so you can
download the bit that was pulled from the ModuleManager.ConfigCache by
Cosmonaut Crash.

https://www.dropbox.com/s/p9us4eu9p9iv22f/HG-3%20Troubleshooting.cfg?dl=0

**Comment by PhineasFreak** [comment]
@Ash19256 let me take a look at the configs and i'll get back to you.

BTW, for the plume config do not worry, it is not required per se and it can be easily added later.

**Comment by Ash19256** [comment]
Alright, so that should fix the Test Flight thing. Also, added a temporary Real Plume config courtesy of Cosmonaut Crash on YouTube. I'll just get this all compiled and off to him for more testing prior to an official merging of this pull request.

**Comment by Ash19256** [comment]
@PhineasFreak is this good now? It has a plume config (albeit not necessarily the most fitting one for the vacuum variant), and the TestFlight stuff should be working.

**Comment by PhineasFreak** [comment]
@Ash19256 the "ModuleEnginesRF" fields are not needed anymore (they are patched automatically) but otherwise it is good!

**Comment by Ash19256** [comment]
Fixed that. Looks like this is ready for merging otherwise.

**Comment by NathanKell** [comment]
Does this already config the SHIP version? https://github.com/Saabstory88/RO_Extended

**Comment by Ash19256** [comment]
Uh.. no. I'll get on that right away.

**Comment by Ash19256** [comment]
Right, that should add SHIP compatibility to this pull request. I'll see what I can do about testing it, but otherwise it should be fine.

**Comment by PhineasFreak** [comment]
@Ash19256 discard my initial comment, it did not work as i expected to. I was talking about these two lines:

https://github.com/Ash19256/RealismOverhaul/blob/42907f86c9535468f44278523bf085c789f373d5/GameData/RealismOverhaul/RO_SuggestedMods/SHIP/Engines.cfg#L18-L19

**Comment by Ash19256** [comment]
Oh, okay. I wasn't sure which lines you were talking about. Those are still in the SHIP/Engines.cfg file.

**Comment by PhineasFreak** [comment]
Yea, it was a bit of a brainfart...

**Comment by Ash19256** [comment]
Out of curiosity, is there a tutorial or something for making a RP-0 PR for something like this? Is it something that I should only do after this PR gets approved/merged or what? Also, is there a How To guide type thing someone could link me to if it's okay to make a pull request to add this into the RP-0 tech tree at this time?

**Comment by NathanKell** [comment]
RP-0 needs to be made aware, yeah--best bet is to talk to @pap1723 about how to integrate them into the new tree, since the methodology has changed significantly from the current release build's way.

**Comment by Ash19256** [comment]
So should I just wait for @pap1723 to make an appearance?

**Comment by pap1723** [comment]
I have appeared!

@Ash19256 What that means is that the parts were not priced and place on the Tech Tree. So RP-0 will automatically flag them as Non-RP0. Do you have real-world pricing on these parts?

**Comment by Ash19256** [comment]
@pap1723 Unfortunately no, I've only been able to find 3 sources, none of which list a unit cost. Considering the HG-3 and it's seal level variant were never actually built IRL (it was, as far as the sources I've found have indicated, an on-paper project only, with no prototypes built), that's not really surprising. Considering the time period and other engines available at the time, I'd imagine that it'd be probably somewhere in the same price range (probably around 5-10% more expensive) than the J-2S, which my sources indicate it's a contemporary of. Take into account the higher chamber pressure making manufacturing more expensive, and making it cost more than the roughly similarly sized J-2S makes sense to me. But ultimately, the price would be a best guesstimate, simply because no IRL price for the engines can be found.

**Comment by NathanKell** [comment]
If memory serves the HG-3 was, like, halfway between J-2 and RS-25*, so I'd suggest a price somewhere between them rather than so close to J-2, maybe?

*RS-25 was developed using a lot of the techniques used for HG-3, IIRC.

**Comment by Ash19256** [comment]
Yeah, that sounds vaguely right. Although, based on it's capabilities, I'd lean more towards the J-2 side of things than the RS-25 side of things, mostly because the engine only produces around 27% more thrust for the same fuel consumption. That being said, it's ultimately up to @pap1723 and the others, so I'll go along with whatever they decide the price should be.

**Comment by Ash19256** [comment]
Right, with this recent update this PR should be compatible with the SHIP mod. All that's needed now is to figure out RP-0 compatibility.

**Comment by NathanKell** [comment]
Great! RP-0 stuff is done on RP-0 side so that's all set! :)

**Pull request merged**